src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -56,44 +59,98 @@ class ZenStackDefinition(SQLModel, table=True):
     )
 
 
-class ZenConfig(SQLModel, table=True):
-    """ "Singleton" Table with only one row storing config."""
-
-    create_time: Optional[dt.datetime] = Field(
-        default_factory=dt.datetime.now, primary_key=True
-    )
-    version: str
-
-
 class SqlStackStore(BaseStackStore):
     """Repository Implementation that uses SQL database backend"""
 
-    def __init__(self, url: str, *args: Any, **kwargs: Any) -> None:
-        """Create a new SqlStackStore.
+    def initialize(
+        self,
+        url: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> "SqlStackStore":
+        """Initialize a new SqlStackStore.
 
         Args:
             url: odbc path to a database.
             args, kwargs: additional parameters for SQLModel.
+
+        Returns:
+            The initialized stack store instance.
         """
 
         logger.debug("Initializing SqlStackStore at %s", url)
+        self._url = url
+
+        local_path = self.get_path_from_url(url)
+        if local_path:
+            fileio.create_dir_recursive_if_not_exists(str(local_path.parent))
+
         self.engine = create_engine(url, *args, **kwargs)
         SQLModel.metadata.create_all(self.engine)
         with Session(self.engine) as session:
-            if not session.exec(select(ZenConfig)).first():
-                session.add(ZenConfig(version=__version__))
             if not session.exec(select(ZenUser)).first():
                 session.add(ZenUser(id=1, name="LocalZenUser"))
             session.commit()
 
+        super().initialize(url, *args, **kwargs)
+
+        return self
+
     # Public interface implementations:
 
     @property
-    def version(self) -> str:
-        """Get the ZenML version."""
+    def type(self) -> StoreType:
+        """The type of stack store."""
+        return StoreType.SQL
+
+    @property
+    def url(self) -> str:
+        """URL of the repository."""
+        if not self._url:
+            raise RuntimeError(
+                "SQL stack store has not been initialized. Call `initialize` "
+                "before using the store."
+            )
+        return self._url
+
+    @staticmethod
+    def get_path_from_url(url: str) -> Optional[Path]:
+        """Get the local path from a URL, it points to a local sqlite file.
+
+        Args:
+            url: The URL to get the path from.
+
+        Returns:
+            The path extracted from the URL, or None, if the URL does not
+            point to a local sqlite file.
+        """
+        if not SqlStackStore.is_valid_url(url):
+            raise ValueError(f"Invalid URL for SQL store: {url}")
+        if not url.startswith("sqlite:///"):
```

Comment:
I'm not sure I understand the difference between this and the check 2 lines above. how can it be a valid url but not start with this string? Furthermore, why return None in this case?

Reply:
The SqlStackStore check is borrowed from SQLAlchemy: it checks that it is a valid SQLAlchemy URL, which can take many forms for the many drivers and SQL databases that it supports. This function only applies to the SQLite case that stores the DB information inside a local file. All other cases are non-local and don't have an associated local path.